app/org/sagebionetworks/bridge/cache/CacheProvider.java

```diff
@@ -26,41 +34,72 @@ public void setJedisOps(JedisOps jedisOps) {
         this.jedisOps = jedisOps;
     }
 
-    public void setUserSession(String key, UserSession session) {
+    public void setUserSession(final UserSession session) {
+        checkNotNull(session);
+        List<Object> results = null;
         try {
-            String ser = BridgeObjectMapper.get().writeValueAsString(session);
-            String redisKey = RedisKey.SESSION.getRedisKey(key);
-            String result = jedisOps.setex(redisKey, BridgeConstants.BRIDGE_SESSION_EXPIRE_IN_SECONDS, ser);
+            final String userId = session.getUser().getId();
+            final String userKey = RedisKey.USER_SESSION.getRedisKey(userId);
+            final String sessionToken = session.getSessionToken();
+            final String sessionKey = RedisKey.SESSION.getRedisKey(sessionToken);
+            final String ser = BridgeObjectMapper.get().writeValueAsString(session);
+            results = jedisOps.getTransaction()
+                    .setex(userKey, BridgeConstants.BRIDGE_SESSION_EXPIRE_IN_SECONDS, sessionToken)
```

Comment:
Indeed the code that does all this is in `AuthenticationController`:
```
    public Result signIn() throws Exception {
        UserSession session = getSessionIfItExists();
        if (session != null) {
            setSessionToken(session.getSessionToken());
            return okResult(new UserSessionInfo(session));
        }
        try {
            JsonNode json = requestToJSON(request());
            SignIn signIn = parseJson(request(), SignIn.class);
            Study study = getStudyOrThrowException(json);
            session = authenticationService.signIn(study, signIn);
            setSessionToken(session.getSessionToken());
            Result result = okResult(new UserSessionInfo(session));
            return result;
        } catch(ConsentRequiredException e) {
            setSessionToken(e.getUserSession().getSessionToken());
            throw e;
        }
    }
```

I believe we're already testing them.

What's new in my change is that session.getUser() is null or user.getId() is null.  They should be exceptions. What I can do is checking on them and throw exceptions.